packages/vite-plugin-cloudflare/playground/assets/src/index.ts

```diff
@@ -0,0 +1,57 @@
+import html from "./index.html?raw";
+import importedImage from "./imported-image.svg";
+import importedText from "./imported-text.txt?url";
+
+interface Env {
+	ASSETS: Fetcher;
+}
+
+class ElementHandler {
+	element(element: Element) {
+		element.setInnerContent("Modified content");
+	}
+}
+
+export default {
+	async fetch(request, env) {
+		const { origin, pathname } = new URL(request.url);
+
+		switch (pathname) {
+			case "/public-directory-asset": {
+				const response = await env.ASSETS.fetch(
+					new URL("/public-image.svg", origin)
```

Comment:
You can't just use any random `origin` here because the URL will pass through Vite's `hostCheckMiddleware`, which checks against the [`server.allowedHosts`](https://vite.dev/config/server-options.html#server-allowedhosts) option. Best thing is to use the `origin` from the incoming request URL. I think this is what we should do in examples.

Reply:
Also, the advantage of using `new URL()` is that the `base` (second parameter) will be ignored if the `url` includes an `origin`. This means that if importing an asset returns a data URI, it can be used without the user having to know it's not a regular URL.